tools/scripts/cn_metrics.py

```diff
@@ -94,14 +94,14 @@ def full_tree(ybuf: Optional[OrderedDict[str, Any]], opt):
         #
         for kvset in node["kvsets"]:
             index = kvset.pop("index")
-            print(f"k {loc['level']},{loc['offset']},{index:<2} ", end="")
+            line = f"k {loc['level']},{loc['offset']},{index:<2}"
 
             col = 0;
             for key, val in kvset.items():
                 width = widthv[col]
                 col += 1
-                print(f"{key} {val:<{width}} ", end="")
-            print()
+                line += f" {key} {val:<{width}}"
+            print(line)
```

Comment:
The say join() is much faster than +=, but if I write something like   line = line.join(f" {key} {val}")   then it seems to go into an infinite loop eating up all the memory on the system (yes, all 512G).

Reply:
I think you have the arguments reversed for .join() :)

Reply:
wait, wouldn't that render the line in the reverse order i want???

Reply:
Make a list of text segments and do one join at the end when you print.

Reply:
Cool, that works.  But it doesn't seem any faster than line += ...   Maybe if I had much larger data set...